content/articles/securing-mongodb-in-linux/index.md

```diff
@@ -0,0 +1,206 @@
+```
+The output will display default databases, and any user can still find important data in one system. To end this situation, we add an administrative user by connecting to the database. To start connecting to the database, we call an admin user to the database using the command below:
+
+```bash
+use admin
+```
+Since the MongoDB environment comes installed with several JSON-based methods used in managing the database, we will use the javascript-based shell method that is used to create a user. The `db.createUser` method is used to add new users to the database. To run the `db.createUser` method, we use the command:
+
+```bash
+db.createUser({user, getLastError})
+#field user contains the documents with authentication information about the user.
+#getlastError can be optional, writes and checks if the user exists.
+```
+The method requires users to create a new user, if exists, an error message will return showing that the database already exists. when creating the user, you need to specify login credentials, passwords, and roles of the user to the database. As we have stated before, MongoDB stores data in JSON documents that hold data in its fields. For a JSON document to execute without any error message, one need to use curly braces to start and end the document created as shown below:
+
+```bash
+#note that without closing parenthesis MongoDB will still not run.
+    use admin
+    db.createUser({
+        user: "AndrewKulundeng'",
+        pwd: "xandrew12@12",#enter roles, closing braces and parenthis.
+        role: [{role:"userAdminAnyDatabase", dp:"admin"}]})
+```
+The method executes properly to prompt the user to input password as shown:
+
+```bash
+*enter password used when creating the database, press enters to execute.*
+Enter password: xandrew12@12
+Output
+Successfully added user: {
+     user : "AndrewKulundeng'",
+     roles : [
+        {
+            "role" : "userAdminAnyDatabase",
+            "db" : "admin"
+        }
+      
+    ]
+}
+```
+Then exist MongoDB after the operation. Users will be allowed to enter details after authentication is enabled and the MongoDB daemon restarts. we can as well restart using the command below:
+```bash
+mongod --auth --dbpath="C:\data\mongo" 0r
+#press enters key to execute.
+$sudo systemctl restart mongodb
+```
+### Turn on authentication
+To make authentication to enable, edit `mongod.conf files` as we have done before then restart your mongo service. This will still allow users to access the database without varifying for passwords. They will not be able to make changes to the database since they do not have the correct username and password. Open the configuration file using the command below and move down to the security. Remove the #sign to the security and add authorization colon and set it to enable as shown:
+
+```bash
+usr/etc/mongod.conf
+security:
+```
+Then add authentication parameter and set it to enable. Note that authorization is indented it will look like this:

```

Comment:
"Note that authorization is indented it will look like this"? This sentence structure isn't correct